File: src/main/java/no/nav/melosys/eessi/repository/FagsakRinasakKoblingRepository.java
package no.nav.melosys.eessi.repository;

import java.util.List;
import java.util.Optional;
import no.nav.melosys.eessi.models.FagsakRinasakKobling;
import org.springframework.data.jpa.repository.JpaRepository;

public interface FagsakRinasakKoblingRepository extends JpaRepository<FagsakRinasakKobling, Long> {

    List<FagsakRinasakKobling> findAllByGsakSaksnummer(Long gsakSaksnummer);

    Optional<FagsakRinasakKobling> findByRinaSaksnummer(String rinaSaksnummer);

    void deleteByRinaSaksnummer(String rinaSaksnummer);
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/repository/FagsakRinasakKoblingRepository.kt
package no.nav.melosys.eessi.repository

import no.nav.melosys.eessi.models.BucType
import no.nav.melosys.eessi.models.FagsakRinasakKobling
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface FagsakRinasakKoblingRepository : JpaRepository<FagsakRinasakKobling, Long> {
    fun findAllByGsakSaksnummer(gsakSaksnummer: Long): MutableList<FagsakRinasakKobling>

    fun findByRinaSaksnummer(rinaSaksnummer: String): Optional<FagsakRinasakKobling>

    fun deleteByRinaSaksnummer(rinaSaksnummer: String)

    fun findAllByGsakSaksnummerAndBucType(gsakSaksnummer: Long, bucType: BucType): MutableList<FagsakRinasakKobling>
}
